docs/core/compatibility/serialization/9.0/binaryformatter-removal.md

```diff
@@ -0,0 +1,48 @@
+---
+title: "Breaking change: In-box BinaryFormatter is unusable"
+description: Learn about the .NET 9 breaking change in serialization where the in-box BinaryFormatter implementation throws exceptions and can't be used successfully.
+ms.date: 08/06/2024
+---
+# In-box BinaryFormatter is unusable
+
+The "in box" <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter> implementation now throws exceptions at run time in all cases. This is the final stage of the [BinaryFormatter obsoletion plan](https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md).
+
+## Previous behavior
+
+You could construct a <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter> instance and use it to serialize and deserialize payloads.
+
+## New behavior
+
+Starting in .NET 9, the in-box <xref:System.Runtime.Serialization.Formatters.Binary.BinaryFormatter> implementation throws exceptions on use, even with the settings that previously enabled its use. Those settings are also removed.
+
+## Version introduced
+
+.NET 9 Preview 6
+
+## Type of breaking change
+
+This change is a [behavioral change](../../categories.md#behavioral-change).
+
+## Reason for change
+
+`BinaryFormatter` is an insecure format and the cause of many security bugs. Removing it from the framework increases the overall safety of .NET.
+
+## Recommended action
```

Comment:
We should probably link to the BinaryFormatter migration topic, but that's not merged yet (#41564).